File: testsuites/code-marathon/challenges/fibonacci/test_runner.c
#include <stdio.h>

extern int fibonacci(int n);

static int tests_run = 0;
static int tests_failed = 0;

#define TEST(name, expr) do { \
    tests_run++; \
    if (!(expr)) { \
        printf("  \u2717 %s\n", name); \
        tests_failed++; \
    } else { \
        printf("  \u2713 %s\n", name); \
    } \
} while(0)

int main(void) {
    printf("[Fibonacci]\n");
    TEST("fibonacci(0)", fibonacci(0) == 0);
    TEST("fibonacci(1)", fibonacci(1) == 1);
    TEST("fibonacci(10)", fibonacci(10) == 55);
    TEST("fibonacci(20)", fibonacci(20) == 6765);
    TEST("fibonacci(30)", fibonacci(30) == 832040);

    printf("\n%d tests, %d passed, %d failed\n", tests_run, tests_run - tests_failed, tests_failed);
    return tests_failed > 0 ? 1 : 0;
}

File: testsuites/code-marathon/challenges/fibonacci_py/test_runner.py
import sys
sys.path.insert(0, '/workspace')

from fibonacci import fibonacci

tests_run = 0
tests_passed = 0

def test(name, expr):
    global tests_run, tests_passed
    tests_run += 1
    if expr:
        tests_passed += 1
        print(f"  \u2713 {name}")
    else:
        print(f"  \u2717 {name}")

print("[Fibonacci (Python)]")
test("fibonacci(0)", fibonacci(0) == 0)
test("fibonacci(1)", fibonacci(1) == 1)
test("fibonacci(10)", fibonacci(10) == 55)
test("fibonacci(20)", fibonacci(20) == 6765)
test("fibonacci(30)", fibonacci(30) == 832040)

print(f"\n{tests_run} tests, {tests_passed} passed, {tests_run - tests_passed} failed")
sys.exit(0 if tests_passed == tests_run else 1)
